src/xcode/ENA/ENA/Source/Services/Exposure Submission/__test__/ExposureSubmissionCheckinTests.swift

```diff
@@ -60,6 +104,41 @@ class ExposureSubmissionCheckinTests: CWATestCase {
 		XCTAssertEqual(preparedCheckins[4].transmissionRiskLevel, 8)
     }
 
+	func testCheckinProtectedReportsPreparation() throws {
+		let service = MockExposureSubmissionService()
+		let appConfig = CachedAppConfigurationMock.defaultAppConfiguration
+
+		let traceLocationId: Data = try XCTUnwrap(Data(base64Encoded: "m686QDEvOYSfRtrRBA8vA58c/6EjjEHp22dTFc+tObY="))
+
+		let checkin = Checkin.mock(
+			traceLocationId: traceLocationId,
+			checkinStartDate: try XCTUnwrap(Calendar.current.date(byAdding: .day, value: -20, to: Date())),
+			checkinEndDate: Date()
+		)
+
+		// process checkins
+		let checkinProtectedReports = service.preparedCheckinProtectedReportsForSubmission(
+			checkins: [checkin],
+			appConfig: appConfig,
+			symptomOnset: .daysSinceOnset(0)
+		)
+
+		XCTAssertEqual(checkinProtectedReports.count, 5)
+
+		let riskLevels = [
+			riskLevel(for: checkinProtectedReports[0], traceLocationId: traceLocationId),
+			riskLevel(for: checkinProtectedReports[1], traceLocationId: traceLocationId),
+			riskLevel(for: checkinProtectedReports[2], traceLocationId: traceLocationId),
+			riskLevel(for: checkinProtectedReports[3], traceLocationId: traceLocationId),
+			riskLevel(for: checkinProtectedReports[4], traceLocationId: traceLocationId)
+		]
+
+		XCTAssertEqual(riskLevels.filter { $0 == 4 }.count, 1)
+		XCTAssertEqual(riskLevels.filter { $0 == 6 }.count, 1)
+		XCTAssertEqual(riskLevels.filter { $0 == 7 }.count, 1)
+		XCTAssertEqual(riskLevels.filter { $0 == 8 }.count, 2)
```

Comment:
I'm wondering if it makes sense to sort them by the `startIntervalNumber` again in this test and then compare them directly like this to ensure that the transmission risk levels are applied in the correct order

```suggestion
		XCTAssertEqual(riskLevel(for: checkinProtectedReports[0], traceLocationId: traceLocationId), 4)
		XCTAssertEqual(riskLevel(for: checkinProtectedReports[1], traceLocationId: traceLocationId), 6)
		XCTAssertEqual(riskLevel(for: checkinProtectedReports[2], traceLocationId: traceLocationId), 7)
		XCTAssertEqual(riskLevel(for: checkinProtectedReports[3], traceLocationId: traceLocationId), 8)
		XCTAssertEqual(riskLevel(for: checkinProtectedReports[4], traceLocationId: traceLocationId), 8)
```

Reply:
done